wls-wahlvorbereitung-service/src/test/java/de/muenchen/oss/wahllokalsystem/wahlvorbereitungservice/service/eroeffnungsuhrzeit/EroeffnungsUhrzeitSecurityTest.java

```diff
@@ -0,0 +1,147 @@
+        }
+
+        @ParameterizedTest(name = "{index} - {1} missing")
+        @MethodSource("getAuthoritiesVariations")
+        void anyMissingAuthorityCausesFail(final ArgumentsAccessor argumentsAccessor) {
+            SecurityUtils.runAs(argumentsAccessor.get(0, String[].class));
+
+            val wahlbezirkID = "wahlbezirkID";
+            Mockito.when(bezirkIDPermissionEvaluator.tokenUserBezirkIdMatches(Mockito.eq(wahlbezirkID), Mockito.any())).thenReturn(true);
+
+            Assertions.assertThatThrownBy(() -> unitUnderTest.getEroeffnungsUhrzeit(wahlbezirkID))
+                    .isInstanceOf(AccessDeniedException.class);
+        }
+
+        private static Stream<Arguments> getAuthoritiesVariations() {
+            return SecurityUtils.buildArgumentsForMissingAuthoritiesVariations(Authorities.ALL_AUTHORITIES_GET_EROEFFNUNGSUHRZEIT);
+        }
+    }
+
+    @Nested
+    class SetEroeffnungsUhrzeit {
+
+        @Test
+        void accessGranted() {
+            SecurityUtils.runAs(Authorities.ALL_AUTHORITIES_POST_EROEFFNUNGSUHRZEIT);
+
+            val wahlbezirkID = "wahlbezirkID";
+            val modelToSet = new EroeffnungsUhrzeitModel(wahlbezirkID, LocalDateTime.now());
+
+            Mockito.when(bezirkIDPermissionEvaluator.tokenUserBezirkIdMatches(Mockito.eq(wahlbezirkID), Mockito.any())).thenReturn(true);
+
+            Assertions.assertThatNoException().isThrownBy(() -> unitUnderTest.setEroeffnungsUhrzeit(modelToSet));
+        }
+
+        @Test
+        void bezirkIDPermissionEvaluatorFailed() {
+            SecurityUtils.runAs(Authorities.ALL_AUTHORITIES_POST_EROEFFNUNGSUHRZEIT);
+
+            val wahlbezirkID = "wahlbezirkID";
+            val modelToSet = new EroeffnungsUhrzeitModel(wahlbezirkID, LocalDateTime.now());
+
+            Mockito.when(bezirkIDPermissionEvaluator.tokenUserBezirkIdMatches(Mockito.eq(wahlbezirkID), Mockito.any())).thenReturn(false);
+
+            Assertions.assertThatException().isThrownBy(() -> unitUnderTest.setEroeffnungsUhrzeit(modelToSet))
+                    .isInstanceOf(AccessDeniedException.class);
+        }
+
+        @Test
+        void accessDeniedOnServiceAuthorityMissing() {
+            SecurityUtils.runAs(Authorities.ALL_AUTHORITIES_REPO_EROEFFNUNGSUHRZEIT);
+
+            val wahlbezirkID = "wahlbezirkID";
+            val modelToSet = new EroeffnungsUhrzeitModel(wahlbezirkID, LocalDateTime.now());
+
+            Mockito.when(bezirkIDPermissionEvaluator.tokenUserBezirkIdMatches(Mockito.eq(wahlbezirkID), Mockito.any())).thenReturn(true);
+
+            Assertions.assertThatThrownBy(() -> unitUnderTest.setEroeffnungsUhrzeit(modelToSet))
+                    .isInstanceOf(AccessDeniedException.class);
+        }
+
+        @Test
+        void wlsExceptionOnRepoWriteAuthorityMissing() {
+
+            String[] ALL_AUTHORITIES_POST_EROEFFNUNGSUHRZEIT_WHITHOUT_REPO_WRITE = Arrays.stream(Authorities.ALL_AUTHORITIES_POST_EROEFFNUNGSUHRZEIT)
```

Comment:
optionale Vorschlag für die Einheitlichkeit

```suggestion
            val ALL_AUTHORITIES_POST_EROEFFNUNGSUHRZEIT_WHITHOUT_REPO_WRITE = Arrays.stream(Authorities.ALL_AUTHORITIES_POST_EROEFFNUNGSUHRZEIT)
```